pkg/servechi/servechi.go

```diff
@@ -0,0 +1,208 @@
+package servechi
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/Contrast-Security-OSS/go-test-bench/internal/common"
+	"github.com/Contrast-Security-OSS/go-test-bench/internal/injection/cmdi"
+	"github.com/Contrast-Security-OSS/go-test-bench/internal/injection/sqli"
+	"github.com/Contrast-Security-OSS/go-test-bench/internal/pathtraversal"
+	"github.com/Contrast-Security-OSS/go-test-bench/internal/ssrf"
+	"github.com/Contrast-Security-OSS/go-test-bench/internal/unvalidated"
+	"github.com/Contrast-Security-OSS/go-test-bench/internal/xss"
+	"github.com/go-chi/chi"
+)
+
+// Pd is unchanging parameter data shared between all routes.
+var Pd = common.ConstParams{
+	Year:      2022,
+	Logo:      "https://cdn.jsdelivr.net/gh/go-chi/chi@master/_examples/chi.svg",
+	Framework: "Chi",
+}
+
+func newHandler(v common.Route) func(w http.ResponseWriter, r *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		log.Println(v.Name, r.URL.Path)
+
+		mode := chi.URLParam(r, "mode")
+		for _, s := range v.Sinks {
+			if chi.URLParam(r, "sink") != s.URL {
+				continue
+			}
+			mode := common.Safety(mode)
+			switch mode {
+			case common.NOOP, common.Safe, common.Unsafe:
+				// valid modes
+			default:
+				// invalid
+				w.WriteHeader(http.StatusNotFound)
+				return
+			}
+
+			in := common.GetUserInput(r)
+			data, mime, status := s.Handler(mode, in, httpHandlerPair{w, r})
+			if len(data) == 0 {
+				// don't unconditionally write response, as it can result in
+				// - an error log of "http: superfluous response.WriteHeader call"
+				return
+			}
+			w.WriteHeader(status)
+			if len(mime) == 0 {
+				mime = "text/plain"
+			}
+			w.Header().Set("Content-Type", mime)
+			w.Header().Set("Cache-Control", "no-store") //makes development a whole lot easier
+			fmt.Fprint(w, data)
+			return
+		}
+		// does not match any sink or the main page
+		w.WriteHeader(http.StatusNotFound)
+	}
+}
+
+func add(router chi.Router, rt common.Route) {
+	for _, s := range rt.Sinks {
+		if s.Handler == nil {
+			var err error
+			s.Handler, err = common.GenericHandler(s)
+			if err != nil {
+				log.Fatal(err)
+			}
+		}
+	}
+	// requests to main page without trailing slash are redirected
+	// with trailing slash included to mimic behavior of other frameworks
```

Comment:
minor: Is the only other framework we're mimicking here julienschmidt? I ran a quick test locally to register on `rt.Base` instead of `rt.Base+"/"` and julienschmidt and chi both passed the cmd/exercise tests and routed correctly via the navbar in Chrome. 

If this redirect is only to align with julienschmidt, I think it will make maintenance easier if we update both frameworks to register routes without a trailing slash instead of expecting them to redirect (or patching on a redirect like we needed to do here).

Reply:
This is also the behavior I observed for the standard router as well.